src/components/questionnaire-builder/Question/index.tsx

```diff
@@ -98,391 +102,386 @@ interface DropData {
             onSelectChange(questionId, !selected);
         }
-    }
+    }, [data, onSelectChange, selected]);
 
-    handleExpandClick = () => {
-        const {
-            data: {
-                id: questionId,
-            },
-            expanded,
-            onExpandChange,
-        } = this.props;
+    const handleExpandClick = useCallback(() => {
+        const { id: questionId } = data;
 
         if (onExpandChange) {
             onExpandChange(questionId, !expanded);
         }
-    }
-
-    public render() {
-        const {
-            selected,
-            className,
-            data,
-            readOnly,
-            disabled,
-            expanded,
-            onSelectChange,
-            onExpandChange,
-            copyDisabled,
-            onCopy,
-
-            onArchive,
-            onUnarchive,
-            onEditButtonClick,
-            onDelete,
-            onClone,
-            onCopyFromDrop,
-            onAddButtonClick,
-
-            searchValue,
-        } = this.props;
-
-        const {
-            type,
-            title,
-            crisisTypeDetail,
-            enumeratorSkillDisplay,
-            dataCollectionTechniqueDisplay,
-            importanceDisplay,
-            requiredDuration,
-            isArchived,
-            attributeTitle,
-        } = data;
-
-        return (
-            <div
-                className={_cs(className, styles.questionContainer)}
-                draggable={isDefined(onCopy)}
-                onDragStart={this.handleDragStart}
-            >
-                <div className={styles.topContainer}>
-                    {onSelectChange && (
-                        <Checkbox
-                            className={styles.checkbox}
-                            checkIconClassName={styles.checkIcon}
-                            value={selected}
-                            onChange={this.handleCheckboxClick}
-                        />
-                    )}
-                    <div className={styles.question}>
-                        <div className={styles.brief}>
-                            <div className={styles.iconContainer}>
-                                <img
-                                    className={styles.icon}
-                                    src={(type ? iconMap[type] : undefined)}
-                                    alt={type}
+    }, [onExpandChange, expanded, data]);
+
+    const {
+        type,
+        title,
+        name,
+        moreTitles,
+        crisisTypeDetail,
+        enumeratorSkillDisplay,
+        dataCollectionTechniqueDisplay,
+        importanceDisplay,
+        requiredDuration,
+        isArchived,
+        attributeTitle,
+    } = data;
+
+    const moreTitlesList: LanguageTitle[] = useMemo(() => {
+        if (isNotDefined(moreTitles)) {
+            return [];
+        }
+        return mapToList(
+            moreTitles,
+            (d: string, k) => {
+                const key = k as string;
```

Comment:
Why are we casting it as string here?

Reply:
K by definition is number | string, and for moreTitles we need string.
I've casted is as string, instead of defining the type.